src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java

```diff
@@ -31,28 +33,29 @@ public ConfigurationFactory() {
   }
 
   /**
-   * @param fileToScan the {@link File} to analyze.
+   * @param inputFile the {@link InputFile} to analyze.
    * @return the {@link Configuration} responsible for the project owning the given {@link File}.
    */
-  public static Configuration get(File fileToScan) {
+  public static Configuration get(InputFile inputFile) {
```

Comment:
Actually InputFile is a very specific/proprietary API of sonarqube. To finally make use of this you internally convert it to a path. Could you use `Path` already here and do the conversion to `Path` in `DevonArchitectureComponentCheck` to keep this class portable. We can then test it easily without the need to mock sonarqube `InputFile`.

Reply:
The thing is that methods that return a `Path` are marked as deprecated as well. What I can do is get the path in a string via `context.getInputFile().uri().getPath()` and then just make the `public static Configuration get()` method take a String as parameter.

Reply:
I refactored the cases where I used InputFile to String now and adjusted the code accordingly. Tests are fine and logs show the same output as before.